lib/files.ncl

```diff
@@ -0,0 +1,63 @@
+let nix = import "./nix-interop/nix.ncl" in
+{
+  File = {
+    target
+      | doc m%"
+        The file to write to.
+        If null, defaults to the attribute name of the file.
+      "%
+      | nix.derivation.NullOr String
+      | default
+      = null,
+    content
+      | doc m%"
+        The content of the file.
+      "%
+      | nix.derivation.NixString,
+    materialisation_method : [| 'Symlink, 'Copy |]
+      | doc m%"
+        How the file should be materialized on-disk.
```

Comment:
I think for our primary use case we care more about representation in Git repo, not on disk.

Reply:
It is, but I'll save these considerations for when we mention `.gitignore` as it'll be more relevant there (and we don't know anything about Git yet)

Reply:
Good point. Yes, we should keep it somewhat consistent.